### DISCUSSION TITLE
Email templates

### DISCUSSION THREAD

**Discussion by jimafisk** [pocketbase/pocketbase#57] [Q&A] [answered]
There appears to be system emails for common actions like:
- user verification
- user password reset
- user email change confirmation

Is there a way to override the default content of these messages? Is that something that should be done at the SMTP provider level? It would be nice if this could be set in the admin dashboard and make use of replacement patterns. Thanks!

**Comment by ganigeorgiev** [comment]
To change the default user emails you can make use of the `OnMailerBeforeUser*` event hooks - https://pocketbase.io/docs/event-hooks/#mailer-hooks.

Initially it was in the admin dashboard (see [earlier mockups](https://app.presentator.io/#/6mote1hf/prototypes/31143/screens/210839?mode=preview)) but I decided to replace it with event hooks because it provides more options for customizations and the behavior is more consistent when the dev uses PocketBase as framework (otherwise you may end up with something that seems configurable in the dashboard, but actually may not be used by the application due to some intercepted hook event).

**Comment by jimafisk** [comment]
The hooks are great, but is there a way to do both so non-developers could customize emails? Not sure how feasible it would be, but if a mailer hook was implemented by the developer could it disable the template override in the dashboard and throw a warning message that it's been overridden by a hook? The mockup you referenced looks great, I could see that being super useful for folks who don't want to use PB as a framework.

**Comment by ganigeorgiev** [comment]
At the moment, there is no way to know what the attached hook listeners do prior execution.
Even if the developer use for example `OnMailerBeforeAdminResetPasswordSend` hook there is no guarantee that this hook is used to send a different email template (eg. it can be used just to notify some external system).

Although I understand that it's not very user friendly as it is now, it is a lot more consistent and easy to reason in the long run.

I'll move the issue to discussions so the other users can join and share their feedback.

**Comment by ganigeorgiev** [comment, marked as answer]
@jimafisk Just to let you know that support to customize the "default" user email templates from the Admin UI was added in [v0.5.0 release](https://github.com/pocketbase/pocketbase/releases/tag/v0.5.0) as part of #275.

_Developers are still able to override the defaults using the mailer hooks._

**Reply by jimafisk** [reply]
I just checked it out on the demo site, this looks awesome @ganigeorgiev! Thanks so much for accommodating this request!

**Reply by YoRHa-A5** [reply]
@ganigeorgiev Is i18n support for email templates something for the roadmap?

**Reply by ganigeorgiev** [reply]
@BunnyTheLifeguard You could check this issue - https://github.com/pocketbase/pocketbase/issues/593.

**Reply by notmorrisjohn** [reply]
was this removed?

**Reply by notmorrisjohn** [reply]
<img width="1346" alt="image" src="https://github.com/user-attachments/assets/25579792-e971-4743-853e-0c0b9b77260d" />

**Reply by ganigeorgiev** [reply]
@notmorrisjohn No, it is still available but with PocketBase v0.23+ the email templates are under each individual auth collection options. For example:

![auth_collection_options](https://i.imgur.com/ZRAIj2T.png)
